isis/src/control/objs/BundleUtilities/BundleObservationVector.cpp

```diff
@@ -117,20 +118,37 @@ namespace Isis {
       bundleImage->setParentObservation(bundleObservation);
 
       // updateo observation number to observation ptr map
-      m_observationNumberToObservationMap.insertMulti(observationNumber,bundleObservation);
+      m_observationNumberToObservationMap.insertMulti(observationNumber, bundleObservation);
 
       // update image serial number to observation ptr map
       m_imageSerialToObservationMap.insertMulti(bundleImage->serialNumber(), bundleObservation);
     }
     else {
       // create new BundleObservation and append to this vector
-      BundleObservation *isisObservation = new BundleObservation(bundleImage,
-                                                                 observationNumber,
-                                                                 instrumentId,
-                                                                 bundleSettings->bundleTargetBody());
 
+      bool isIsisObservation = true;
+
+      // This NULL check is needed solely for the unit test
```

Comment:
How much work would it be to update the unit test and remove these if camera!=NULL checks?

Reply:
I'm not entirely sure. I can give it a try and timebox it?

Reply:
This looks like it's working so how about we merge this and you can put up a separate PR that tries to remove these?